### ISSUE TITLE
Using TensorFlow backend

### ISSUE THREAD

**Issue by danuonuo** [deepfakes/faceswap#1035]
I can't use my GPU. 

============ System Information ============
encoding:            cp936
git_branch:          master
git_commits:         2f15597 requirements.txt - typofix
gpu_cuda:            No global version found. Check Conda packages for Conda Cuda
gpu_cudnn:           No global version found. Check Conda packages for Conda cuDNN
gpu_devices:         GPU_0: GeForce RTX 2060
gpu_devices_active:  GPU_0
gpu_driver:          436.30
gpu_vram:            GPU_0: 6144MB
os_machine:          AMD64
os_platform:         Windows-10-10.0.18362-SP0
os_release:          10
py_command:          C:\Users\Administrator\faceswap/faceswap.py gui
py_conda_version:    conda 4.8.3
py_implementation:   CPython
py_version:          3.7.7
py_virtual_env:      True
sys_cores:           6
sys_processor:       AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD
sys_ram:             Total: 16333MB, Available: 8629MB, Used: 7704MB, Free: 8629MB

=============== Pip Packages ===============
absl-py==0.9.0
astor==0.8.0
blinker==1.4
brotlipy==0.7.0
cachetools==4.1.0
certifi==2020.6.20
cffi==1.14.0
chardet==3.0.4
click==7.1.2
cloudpickle @ file:///tmp/build/80754af9/cloudpickle_1594141588948/work
cryptography==2.9.2
cycler==0.10.0
cytoolz==0.10.1
dask @ file:///tmp/build/80754af9/dask-core_1594156306305/work
decorator==4.4.2
fastcluster==1.1.26
ffmpy==0.2.3
gast==0.2.2
google-auth @ file:///tmp/build/80754af9/google-auth_1594357566944/work
google-auth-oauthlib==0.4.1
google-pasta==0.2.0
grpcio==1.27.2
h5py==2.10.0
idna @ file:///tmp/build/80754af9/idna_1593446292537/work
imageio @ file:///tmp/build/80754af9/imageio_1594161405741/work
imageio-ffmpeg @ file:///home/conda/feedstock_root/build_artifacts/imageio-ffmpeg_1589202782679/work
joblib @ file:///tmp/build/80754af9/joblib_1594236160679/work
Keras==2.2.4
Keras-Applications @ file:///tmp/build/80754af9/keras-applications_1594366238411/work
Keras-Preprocessing==1.1.0
kiwisolver==1.2.0
Markdown==3.1.1
matplotlib @ file:///C:/ci/matplotlib-base_1592846084747/work
mkl-fft==1.1.0
mkl-random==1.1.1
mkl-service==2.3.0
networkx @ file:///tmp/build/80754af9/networkx_1594377231366/work
numpy==1.18.5
nvidia-ml-py3 @ git+https://github.com/deepfakes/nvidia-ml-py3.git@6fc29ac84b32bad877f078cb4a777c1548a00bf6
oauthlib==3.1.0
olefile==0.46
opencv-python==4.3.0.36
opt-einsum==3.1.0
Pillow @ file:///C:/ci/pillow_1594298234712/work
protobuf==3.12.3
psutil==5.7.0
pyasn1==0.4.8
pyasn1-modules==0.2.7
pycparser @ file:///tmp/build/80754af9/pycparser_1594388511720/work
PyJWT==1.7.1
pyOpenSSL @ file:///tmp/build/80754af9/pyopenssl_1594392929924/work
pyparsing==2.4.7
pyreadline==2.1
PySocks @ file:///C:/ci/pysocks_1594394709107/work
python-dateutil==2.8.1
PyWavelets==1.1.1
pywin32==227
PyYAML==5.3.1
requests @ file:///tmp/build/80754af9/requests_1592841827918/work
requests-oauthlib==1.3.0
rsa==4.0
scikit-image==0.16.2
scikit-learn @ file:///C:/ci/scikit-learn_1592847564598/work
scipy @ file:///C:/ci/scipy_1592916958183/work
six==1.15.0
tensorboard==2.2.1
tensorboard-plugin-wit==1.6.0
tensorflow==1.15.0
tensorflow-estimator==1.15.1
termcolor==1.1.0
threadpoolctl @ file:///tmp/tmp9twdgx9k/threadpoolctl-2.1.0-py3-none-any.whl
toolz==0.10.0
toposort==1.5
tornado==6.0.4
tqdm @ file:///tmp/build/80754af9/tqdm_1593446365756/work
urllib3==1.25.9
Werkzeug==0.16.1
win-inet-pton==1.1.0
wincertstore==0.2
wrapt==1.12.1

============== Conda Packages ==============
# packages in environment at C:\Users\Administrator\MiniConda3\envs\faceswap:
#
# Name                    Version                   Build  Channel
_tflow_select             2.2.0                     eigen  
absl-py                   0.9.0                    py37_0  
astor                     0.8.0                    py37_0  
blas                      1.0                         mkl  
blinker                   1.4                      py37_0  
brotlipy                  0.7.0           py37he774522_1000  
ca-certificates           2020.6.24                     0  
cachetools                4.1.0                      py_1  
certifi                   2020.6.20                py37_0  
cffi                      1.14.0           py37h7a1dbc1_0  
chardet                   3.0.4                 py37_1003  
click                     7.1.2                      py_0  
cloudpickle               1.5.0                      py_0  
cryptography              2.9.2            py37h7a1dbc1_0  
cycler                    0.10.0                   py37_0  
cytoolz                   0.10.1           py37he774522_0  
dask-core                 2.20.0                     py_0  
decorator                 4.4.2                      py_0  
fastcluster               1.1.26           py37h9b59f54_1    conda-forge
ffmpeg                    4.3                  ha925a31_0    conda-forge
ffmpy                     0.2.3                    pypi_0    pypi
freetype                  2.10.2               hd328e21_0  
gast                      0.2.2                    py37_0  
git                       2.23.0               h6bb4b03_0  
google-auth               1.17.2                     py_0  
google-auth-oauthlib      0.4.1                      py_2  
google-pasta              0.2.0                      py_0  
grpcio                    1.27.2           py37h351948d_0  
h5py                      2.10.0           py37h5e291fa_0  
hdf5                      1.10.4               h7ebc959_0  
icc_rt                    2019.0.0             h0cc432a_1  
icu                       58.2                 ha925a31_3  
idna                      2.10                       py_0  
imageio                   2.9.0                      py_0  
imageio-ffmpeg            0.4.2                      py_0    conda-forge
intel-openmp              2020.1                      216  
joblib                    0.16.0                     py_0  
jpeg                      9b                   hb83a4c4_2  
keras                     2.2.4                         0  
keras-applications        1.0.8                      py_1  
keras-base                2.2.4                    py37_0  
keras-preprocessing       1.1.0                      py_1  
kiwisolver                1.2.0            py37h74a9793_0  
libpng                    1.6.37               h2a8f88b_0  
libprotobuf               3.12.3               h7bd577a_0  
libtiff                   4.1.0                h56a325e_1  
lz4-c                     1.9.2                h62dcd97_0  
markdown                  3.1.1                    py37_0  
matplotlib                3.2.2                         0  
matplotlib-base           3.2.2            py37h64f37c6_0  
mkl                       2020.1                      216  
mkl-service               2.3.0            py37hb782905_0  
mkl_fft                   1.1.0            py37h45dec08_0  
mkl_random                1.1.1            py37h47e9c7a_0  
networkx                  2.4                        py_1  
numpy                     1.18.5           py37h6530119_0  
numpy-base                1.18.5           py37hc3f5095_0  
nvidia-ml-py3             7.352.1                  pypi_0    pypi
oauthlib                  3.1.0                      py_0  
olefile                   0.46                     py37_0  
opencv-python             4.3.0.36                 pypi_0    pypi
openssl                   1.1.1g               he774522_0  
opt_einsum                3.1.0                      py_0  
pathlib                   1.0.1                    py37_2  
pillow                    7.2.0            py37hcc1f983_0  
pip                       20.1.1                   py37_1  
protobuf                  3.12.3           py37h33f27b4_0  
psutil                    5.7.0            py37he774522_0  
pyasn1                    0.4.8                      py_0  
pyasn1-modules            0.2.7                      py_0  
pycparser                 2.20                       py_2  
pyjwt                     1.7.1                    py37_0  
pyopenssl                 19.1.0                     py_1  
pyparsing                 2.4.7                      py_0  
pyqt                      5.9.2            py37h6538335_2  
pyreadline                2.1                      py37_1  
pysocks                   1.7.1                    py37_1  
python                    3.7.7                h81c818b_4  
python-dateutil           2.8.1                      py_0  
python_abi                3.7                     1_cp37m    conda-forge
pywavelets                1.1.1            py37he774522_0  
pywin32                   227              py37he774522_1  
pyyaml                    5.3.1            py37he774522_1  
qt                        5.9.7            vc14h73c81de_0  
requests                  2.24.0                     py_0  
requests-oauthlib         1.3.0                      py_0  
rsa                       4.0                        py_0  
scikit-image              0.16.2           py37h47e9c7a_0  
scikit-learn              0.23.1           py37h25d0782_0  
scipy                     1.5.0            py37h9439919_0  
setuptools                49.2.0                   py37_0  
sip                       4.19.8           py37h6538335_0  
six                       1.15.0                     py_0  
sqlite                    3.32.3               h2a8f88b_0  
tensorboard               2.2.1              pyh532a8cf_0  
tensorboard-plugin-wit    1.6.0                      py_0  
tensorflow                1.15.0          eigen_py37h9f89a44_0  
tensorflow-base           1.15.0          eigen_py37h07d2309_0  
tensorflow-estimator      1.15.1             pyh2649769_0  
termcolor                 1.1.0                    py37_1  
threadpoolctl             2.1.0              pyh5ca1d4c_0  
tk                        8.6.10               he774522_0  
toolz                     0.10.0                     py_0  
toposort                  1.5                        py_3    conda-forge
tornado                   6.0.4            py37he774522_1  
tqdm                      4.47.0                     py_0  
urllib3                   1.25.9                     py_0  
vc                        14.1                 h0510ff6_4  
vs2015_runtime            14.16.27012          hf0eaf9b_3  
werkzeug                  0.16.1                     py_0  
wheel                     0.34.2                   py37_0  
win_inet_pton             1.1.0                    py37_0  
wincertstore              0.2                      py37_0  
wrapt                     1.12.1           py37he774522_1  
xz                        5.2.5                h62dcd97_0  
yaml                      0.2.5                he774522_0  
zlib                      1.2.11               h62dcd97_4  
zstd                      1.4.5                ha9fde0e_0  

================= Configs ==================
--------- .faceswap ---------
backend:                  nvidia

--------- convert.ini ---------

[color.color_transfer]
clip:                     True
preserve_paper:           True

[color.manual_balance]
colorspace:               HSV
balance_1:                0.0
balance_2:                0.0
balance_3:                0.0
contrast:                 0.0
brightness:               0.0

[color.match_hist]
threshold:                99.0

[mask.box_blend]
type:                     gaussian
distance:                 11.0
radius:                   5.0
passes:                   1

[mask.mask_blend]
type:                     normalized
kernel_size:              3
passes:                   4
threshold:                4
erosion:                  0.0

[scaling.sharpen]
method:                   unsharp_mask
amount:                   150
radius:                   0.3
threshold:                5.0

[writer.ffmpeg]
container:                mp4
codec:                    libx264
crf:                      23
preset:                   medium
tune:                     none
profile:                  auto
level:                    auto

[writer.gif]
fps:                      25
loop:                     0
palettesize:              256
subrectangles:            False

[writer.opencv]
format:                   png
draw_transparent:         False
jpg_quality:              75
png_compress_level:       3

[writer.pillow]
format:                   png
draw_transparent:         False
optimize:                 False
gif_interlace:            True
jpg_quality:              75
png_compress_level:       3
tif_compression:          tiff_deflate

--------- extract.ini ---------

[global]
allow_growth:             False

[align.fan]
batch-size:               12

[detect.cv2_dnn]
confidence:               50

[detect.mtcnn]
minsize:                  20
threshold_1:              0.6
threshold_2:              0.7
threshold_3:              0.7
scalefactor:              0.709
batch-size:               8

[detect.s3fd]
confidence:               70
batch-size:               4

[mask.unet_dfl]
batch-size:               8

[mask.vgg_clear]
batch-size:               6

[mask.vgg_obstructed]
batch-size:               2

--------- gui.ini ---------

[global]
fullscreen:               False
tab:                      extract
options_panel_width:      30
console_panel_height:     20
icon_size:                14
font:                     default
font_size:                9
autosave_last_session:    prompt
timeout:                  120
auto_load_model_stats:    True

--------- train.ini ---------

[global]
coverage:                 68.75
mask_type:                none
mask_blur_kernel:         3
mask_threshold:           4
learn_mask:               False
icnr_init:                False
conv_aware_init:          False
reflect_padding:          False
penalized_mask_loss:      True
loss_function:            mae
learning_rate:            5e-05

[model.dfl_h128]
lowmem:                   False

[model.dfl_sae]
input_size:               128
clipnorm:                 True
architecture:             df
autoencoder_dims:         0
encoder_dims:             42
decoder_dims:             21
multiscale_decoder:       False

[model.dlight]
features:                 best
details:                  good
output_size:              256

[model.original]
lowmem:                   False

[model.realface]
input_size:               64
output_size:              128
dense_nodes:              1536
complexity_encoder:       128
complexity_decoder:       512

[model.unbalanced]
input_size:               128
lowmem:                   False
clipnorm:                 True
nodes:                    1024
complexity_encoder:       128
complexity_decoder_a:     384
complexity_decoder_b:     512

[model.villain]
lowmem:                   False

[trainer.original]
preview_images:           14
zoom_amount:              5
rotation_range:           10
shift_range:              5
flip_chance:              50
color_lightness:          30
color_ab:                 8
color_clahe_chance:       50
color_clahe_max_size:     4

**Comment by torzdf** [comment]
You have the CPU version installed.

Easiest fix is to remove your faceswap folder, re-run the installer and make sure you select the "Nvidia" option

**Comment by danuonuo** [comment]
> You have the CPU version installed.
> 
> Easiest fix is to remove your faceswap folder, re-run the installer and make sure you select the "Nvidia" option

 But I chose GPU installation.

**Comment by torzdf** [comment]
That's fine, but I can see from your environment you have the CPU version installed, so follow the instructions and it should resolve the issue

**Comment by danuonuo** [comment]
> That's fine, but I can see from your environment you have the CPU version installed, so follow the instructions and it should resolve the issue

Thank you. I'll try.
